src/lib.rs

```diff
@@ -377,13 +378,12 @@ where
                 &self.map,
                 guard,
             ) {
-                // If `insert` returns false it means the hashmap has been
+                // If `insert` returns retry it means the hashmap has been
                 // resized, we need to try to insert the kvpair again.
-                map_inner::WriteResult::Retry => continue,
-                map_inner::WriteResult::Succ(result) => {
-                    return &result
-                        .unwrap_or(unsafe { kvpair.as_raw().as_ref().unwrap() })
-                        .value;
+                map_inner::InsertResult::Retry => continue,
+                map_inner::InsertResult::Fail(result) => return &result.unwrap().value,
+                map_inner::InsertResult::Succ(_) => {
+                    return unsafe { &kvpair.as_raw().as_ref().unwrap().value };
```

Comment:
clippy restriction lints: `clippy::unwrap_used`, `clippy::undocumented_unsafe_blocks`
```rust
// SAFETY: {{explain why we can deref the raw pointer}}
return unsafe { &(*kvpair.as_raw()).value };
```

Reply:
What is good practice when we are sure that an `Option` is not `None` and we want to use unwrap()?

Reply:
Safe Rust: Just use `.unwrap()` or `.expect("...")` if we don't know how to handle the `None`.
Unsafe Rust: Skip all unnecessary checks, or use `Option::unwrap_unchecked` (since 1.58). `debug_assert!` is also a good supplement. The author must take care of the soundness.